test/python/transpiler/test_swap_mapper.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Test the Swap Mapper pass"""
+
+import unittest
+
+from qiskit.transpiler.passes import SwapMapper
+from qiskit.mapper import Coupling
+from qiskit.dagcircuit import DAGCircuit
+from qiskit import QuantumRegister
+from ..common import QiskitTestCase
+
+
+def _create_dag(instructions):
+    dag = DAGCircuit()
+    qregs = {}
+    basic_elements = {}
+
+    for instruction in instructions:
+        basic_elements[instruction[0]] = len(instruction[1])
+
+        for arg in instruction[1]:
+            if qregs.get(arg[0]) is None:
+                qregs[arg[0]] = []
+            qregs[arg[0]].append(arg[1])
+
+    # dag.add_qreg
+    for name, regs in qregs.items():
+        dag.add_qreg(QuantumRegister(max(regs) + 1, name))
+
+    # dag.add_basis_element
+    for name, amount in basic_elements.items():
+        dag.add_basis_element(name, amount)
+
+    # apply_operation_back
+    for instruction, args in instructions:
+        dag.apply_operation_back(instruction, args)
+
+    return dag
+
+
+class TestSwapMapper(QiskitTestCase):
+    """ Tests the SwapMapper pass."""
+
+    def test_trivial_case(self):
+        """No need to have any swap, the CX are distance 1 to each other
+         q0:--(+)-[U]-(+)-
+               |       |
+         q1:---.-------|--
+                       |
+         q2:-----------.--
+
+         Coupling map: [1]--[0]--[2]
+        """
+        coupling = Coupling({0: [1, 2]})
```

Comment:
We are deprecating the "dict" style of coupling map, in favor of a "list" style. In #1141, @kdk has made a new Coupling constructor based on list. I think we should use that here too, and just stop using dict altogether for consistency.

Reply:
The pass takes a `Coupling` object. It that class keeps the API, the pass should continue working. If the constructor changes, the test should be updated.